addons/purchase/models/account_invoice.py

```diff
@@ -177,11 +188,21 @@ def _anglo_saxon_purchase_move_lines(self, i_line, res):
                             for val_stock_move in valuation_stock_move:
                                 valuation_price_unit_total += abs(val_stock_move.price_unit) * val_stock_move.product_qty
                                 valuation_total_qty += val_stock_move.product_qty
+                            # in Stock Move price unit is in company_currency
                             valuation_price_unit = valuation_price_unit_total / valuation_total_qty
                             valuation_price_unit = i_line.product_id.uom_id._compute_price(valuation_price_unit, i_line.uom_id)
+
+                    line_price_unit_company_currency = i_line.price_unit
                     if inv.currency_id.id != company_currency.id:
-                            valuation_price_unit = company_currency.with_context(date=inv.date_invoice).compute(valuation_price_unit, inv.currency_id, round=False)
-                    if valuation_price_unit != i_line.price_unit and line['price_unit'] == i_line.price_unit and acc:
+                            line_price_unit_company_currency = (
```

Comment:
Are you sure? You convert to `inv.currency_id`. However `valuation_price_unit != line_price_unit_company_currency` is ok since they are both in the order currency. And `i_line.price_unit` is in the invoice currency since it is a `account.invoice.line`, right?

Reply:
@nim-odoo
Woops, I inverted from_currency and to_currency in currency.compute() ; my bad
now we do compare valuation_price_unit and line_price_unit in terms of the company's currency
The data in test case 2 are much better too